### ISSUE TITLE
Type to search appears even there is no match for search term when autoFocus is true.

### ISSUE THREAD

**Issue by rushijagani** [ericgio/react-bootstrap-typeahead#479]
### Version
16.7.0

### Steps to reproduce
```
<AsyncTypeahead
    autoFocus={true}
    {...tProps}
    ref={typeahead => {
        this.typeahead = typeahead;
    }}
    className={search && 'search'}
    onSearch={this.handleSearch}
    onChange={this.handleChange}
    onBlur={this.onBlur}
    onFocus={this.onFocus}

/>
```

```
componentWillReceiveProps(props) {
    const { value } = props;
    if (value !== selected) {
      this.typeahead && this.typeahead.getInstance().focus();
    }
  }
```

When the input field is auto-focused let's say it has already "asdasdasd" search term then it is showing the "Type to search..." ideally it should display "No matches found."
it is working as expected then there is a match search term and results.

After focus it I change anything then the "No matches found." message appears

1. Initially when there are search term and auto-focus is triggered
![Screenshot (34)](https://user-images.githubusercontent.com/51742066/64000943-62e5cf80-cb24-11e9-87bd-6e3b75a990b8.png)

2. if I update the input then it shows the correct message "No matches found."
![Screenshot (35)](https://user-images.githubusercontent.com/51742066/64000991-801a9e00-cb24-11e9-9c26-e1050afd8788.png)

### Expected Behavior

When the typehead is focused and there is already "asdasda" search term with no match results then it should display the No Match message instead of the "Type to search..."

**Comment by ericgio** [comment]
Hi @rushijagani, thanks for filing this issue! I'm assuming you're using `defaultInputValue` to set the initial input text. It looks like I need to set the initial query in the async version when that value is set.

**Comment by ericgio** [comment]
Fixed in v3.4.7
